src/providers/scicat-provider/dao/mongo-dao.ts

```diff
@@ -50,68 +58,43 @@ export class MongoConnector {
    * @param parameters
    * @returns {Promise<any>}
    */
-  public recordsQuery(parameters: any, filter: MongoClient.filter): Promise<any> {
+  public async recordsQuery(parameters: any, filter: any): Promise<any> {
```

Comment:
I think you can now import the filter from 
```
import { Filter } from 'mongodb';
```
so that we can keep type check

Reply:
Fixed. [4485e2e](https://github.com/SciCatProject/oai-provider-service/pull/112/commits/4485e2e94707a02121986363c1282012b0164fea)